planning/behavior_path_planner/src/scene_module/lane_change/util.cpp

```diff
@@ -330,92 +327,62 @@ bool isLaneChangePathSafe(
-  constexpr double check_distance = 100.0;
 
-  // parameters
-  const double time_resolution = ros_parameters.prediction_time_resolution;
-  const double min_thresh = ros_parameters.min_stop_distance;
-  const double stop_time = ros_parameters.stop_time;
-
-  double buffer;
-  double lateral_buffer;
-  if (use_buffer) {
-    buffer = ros_parameters.hysteresis_buffer_distance;
-    lateral_buffer = 0.5;
-  } else {
-    buffer = 0.0;
-    lateral_buffer = 0.0;
-  }
-  double current_lane_check_start_time = 0.0;
-  const double current_lane_check_end_time =
-    ros_parameters.lane_change_prepare_duration + ros_parameters.lane_changing_duration;
-  double target_lane_check_start_time = 0.0;
-  const double target_lane_check_end_time =
-    ros_parameters.lane_change_prepare_duration + ros_parameters.lane_changing_duration;
-  if (!ros_parameters.enable_collision_check_at_prepare_phase) {
-    current_lane_check_start_time = ros_parameters.lane_change_prepare_duration;
-    target_lane_check_start_time = ros_parameters.lane_change_prepare_duration;
+  if (path.points.empty() || target_lanes.empty() || current_lanes.empty()) {
+    return false;
   }
 
+  const double time_resolution = lane_change_parameters.prediction_time_resolution;
+  const auto & lane_change_prepare_duration = lane_change_parameters.lane_change_prepare_duration;
+  const auto & lane_changing_duration = lane_change_parameters.lane_changing_duration;
+  const double current_lane_check_start_time =
+    (!lane_change_parameters.enable_collision_check_at_prepare_phase) ? lane_change_prepare_duration
+                                                                      : 0.0;
+  const double current_lane_check_end_time = lane_change_prepare_duration + lane_changing_duration;
+  double target_lane_check_start_time =
+    (!lane_change_parameters.enable_collision_check_at_prepare_phase) ? lane_change_prepare_duration
+                                                                      : 0.0;
+  const double target_lane_check_end_time = lane_change_prepare_duration + lane_changing_duration;
+  constexpr double ego_predicted_path_min_speed{1.0};
+  const auto vehicle_predicted_path = util::convertToPredictedPath(
+    path, current_twist, current_pose, target_lane_check_end_time, time_resolution, acceleration,
+    ego_predicted_path_min_speed);
+
+  const auto arc = lanelet::utils::getArcCoordinates(current_lanes, current_pose);
+
   // find obstacle in lane change target lanes
   // retrieve lanes that are merging target lanes as well
   const auto target_lane_object_indices =
     util::filterObjectsByLanelets(*dynamic_objects, target_lanes);
 
   // find objects in current lane
+  constexpr double check_distance = 100.0;
   const auto current_lane_object_indices_lanelet = util::filterObjectsByLanelets(
     *dynamic_objects, current_lanes, arc.length, arc.length + check_distance);
+
+  const double lateral_buffer = (use_buffer) ? 0.5 : 0.0;
```

Comment:
the `lateral_buffer` is used to filter out dynamic object in the current lane. 
Assuming the distance of the object to ego current path's line is `d`, then if 

```
d < (0.5 vehicle_width + lateral_buffer)
```

then the object is considered in current lane.

`lateral_distance_threshold` is the threshold for safety check. Assuming distance between ego's polygon and object's polygon is `h`, if 

```
h < lateral_distance_threshold
```

then we will check for the longitudinal distance.

Reply:
Thanks. I understood the implementation, but still not clear about why there are two filtering for the object. Could you explain one situation where these two filters are needed?

Reply:
One of the situation i can think of is for debug purpose, at least this is what my current use case are (I am not sure about the intention of the original author), and it will allow us to have more information about the object or failures.